doc/src/documentation_style_guide.rst

```diff
@@ -0,0 +1,1019 @@
+developers. It also serves as an initial location to point viewers to other
+relevant resources on the web. The SymPy website has basic information on SymPy
+and how to obtain it, as well as examples to advertise it to users, but it does
+not have technical documentation. The source files are located in the SymPy
+`webpage directory <https://github.com/sympy/sympy.github.com>`_. Appropriate
+items for the website are:
+
+* General descriptions of what SymPy and the SymPy community are
+* Explanations/demonstrations of major software features
+* Listings of other major software that uses SymPy
+* Getting started info for users (download and install instructions)
+* Getting started info for developers
+* Where users can get help and support on using SymPy
+* News about SymPy
+
+**SymPy Documentation** https://docs.sympy.org
+
+This is the main place where users go to learn how to use SymPy. It contains a
+tutorial for SymPy as well as technical documentation for all of the modules.
+The source files are hosted in the main SymPy repository in the `doc directory
+<https://github.com/sympy/sympy/tree/master/doc>`_ at and are built using the
+`Sphinx site generator <http://www.sphinx-doc.org/en/master/>`_ and uploaded to
+the docs.sympy.org site automatically. The docs website also contains a built-
+in shell (SymPy Live) that allows users to interactively execute examples.
+There are two primary types of pages that are generated from different source
+files in the docs directory:
+
+* Narrative Pages: reStructuredText files that correspond to manually written
+  documentation pages not present in the Python source code. Examples are the
+  `tutorial RST files
+  <https://github.com/sympy/sympy/tree/master/doc/src/tutorial>`_. In general,
+  if your documentation is not API documentation it belongs in a narrative page.
+* API Documentation Pages: reStructuredText files that contain directives that
+  generate the Application Programming Interface documentation. These are
+  automatically generated from the SymPy Python source code.
+
+**SymPy Source Code** https://github.com/sympy/sympy
+
+Most functions and classes have documentation written inside it in the form of a
+docstring, which explains the function and includes examples called doctests.
+The purpose of these docstrings are to explain the API of that class or
+function. The doctests examples are tested as part of the test suite, so that we
+know that they always produce the output that they say that they do. Here is an
+`example docstring
+<https://github.com/sympy/sympy/blob/b176f6a1d9890b42dc361857c887992315e3d5ad/sympy/functions/elementary/complexes.py#L22-L47>`_.
+Most docstrings are also automatically included in the Sphinx documentation
+above, so that they appear on the SymPy Documentation website. Here is that
+`same docstring
+<https://docs.sympy.org/latest/modules/functions/elementary.html#sympy.functions.elementary.complexes.im>`_
```

Comment:
The docs shouldn't have any direct links to docs.sympy.org. Rather, use the cross reference syntax. I think 

:obj:\`~.im`

should work.

Reply:
Although this could get out of date from the source linked to above. So I wonder what we should do about that.

Reply:
It should be

:obj:\`same docstring <~.im>\`

Reply:
One note is that the purpose of saying this is so people new to the process understand that there is a source file in the code repository and that it gets rendered on the website.

Reply:
Yes I agree this is helpful to have here. As a newcomer I found it helpful to see the same docstring in the source code and rendered on the website so I could start to understand the different places where docstrings can be viewed. I will fix the link to use the cross reference syntax instead. Thanks!